Source/WebCore/html/canvas/CanvasRenderingContext2D.cpp

```diff
@@ -246,4 +246,11 @@ void CanvasRenderingContext2D::drawTextInternal(const String& text, double x, do
     drawTextUnchecked(textRun, x, y, fill, maxWidth);
 }
 
+void CanvasRenderingContext2D::reset()
+{
+    CanvasRenderingContext2DBase::reset();
+
+    setFont("10px sans-serif"_s);
```

Comment:
Seems a bit arbitrary...

Reply:
The font value? It's the default value https://html.spec.whatwg.org/multipage/canvas.html#2dcontext:dom-context-2d-font

Anyway, yeah this kinda looks weird so this is my reasoning: `reset()` requires setting all canvas state to their default value. This is why I have a whole bunch of `setXXX()` in `CanvasRenderingContext2DBase::reset()`. When it comes to font though, those can only be set in `CanvasRenderingContext2DBase`'s children, namely `CanvasRenderingContext2D` and `OffscreenCanvasRenderingContext2D`. This is why those two subclass overrides the definition of `reset()` to reset the font also. Maybe there's a more elegant way to do this, I don't know.

Reply:
Actually I think this is not needed. `CanvasRenderingContext2DBase::reset()` makes the following call:

`m_stateStack.first() = State();`

This will set the state top-of-stack to the default State which has the following initializer: 

```
CanvasRenderingContext2DBase::State::State()
    ...
    , unparsedFont(DefaultFont)
{
}
```